google/cloud/pubsub_v1/gapic/subscriber_client.py

```diff
@@ -1170,6 +1170,11 @@ def streaming_pull(
                 client_info=self._client_info,
             )
 
+            # Wrappers in api-core should not automatically pre-fetch the first
+            # stream result, as this breaks the stream when re-opening it.
+            # https://github.com/googleapis/python-pubsub/issues/93#issuecomment-630762257
```

Comment:
This is purely a performance thing yes? It shouldn't have any impact otherwise?

Reply:
Not sure if I understand the performance part?

The complementing [fix in api-core](https://github.com/googleapis/python-api-core/pull/30/files#diff-d97e81006eaaf29e33270a85aead6aaf) just checks the flag added here and can be used to disable consuming the first stream item on stream creation.

After applying it, the `api-core` will again start behaving the same as in it was before the 1.17.0 release while still preserving [the fix](https://github.com/googleapis/python-api-core/commit/2b103b60ece16a1e1bc98cfda7ec375191a90f75#diff-d97e81006eaaf29e33270a85aead6aaf) that was added for Firestore, but unfortunately broke PubSub in the process.

Reply:
Got it, makes sense. I was wondering what the performance implications of not doing the pre-fetching, but it seems like it's negligible.